egs/librispeech/ASR/RESULTS.md

```diff
@@ -1170,11 +1171,27 @@ export CUDA_VISIBLE_DEVICES="0,1,2,3"
   --start-epoch 0 \
   --num-epochs 90
 # Note: It trains for 90 epochs, but the best WER is at epoch-77.pt
+
+# Train the RNN-LM
```

Comment:
Could you upload the trained RNN-LM model and the training logs to huggingface and place a link to it here? 
By the way, what is the PPL on test-clean of this RNN-LM model?

Reply:
The ppl on test-clean and test-other (from prepare.sh) is 9.970. Do you want test-clean in particular ?

Reply:
Hm... we'll have to look in papers, how people report the LM perplexity.  But normally the perplexity would be per word, not per BPE token.  I.e. you would have to take 9.970 ** { num_words_in_test_set / num_bpe_tokens_in_test_set}

Reply:
BTW, we plan to work on an un-normalized version of RNN-T that will benefit much more, I think, from improved language models.  So the value of this language modeling stuff is more than just the immediate WER gains.  We need to gain some experience with training neural LMs.

Reply:
There are 211138 tokens and 104919 words in test set, so the perplexity per word should be 3.13.

The un-normalized version of RNN-T sounds pretty interesting !

Reply:
I had the formulat the wrong way around, should be ** {num_tokens/num_words}